### DISCUSSION TITLE
Date comparisons with Rego

### DISCUSSION THREAD

**Discussion by peteroneilljr** [open-policy-agent/community#21] [OPA and Rego] [answered]
How to use Rego for date comparisons?

**Comment by peteroneilljr** [comment, marked as answer]
Answer taken from [slack](https://openpolicyagent.slack.com/archives/C1H0ZF6BE/p1632943572100200?thread_ts=1632942075.099300&cid=C1H0ZF6BE) 

```
package play

time_window_min := time.parse_rfc3339_ns("2019-10-22T12:00:00Z")
time_window_max := time.parse_rfc3339_ns("2022-10-22T13:00:00Z")

default allow = false

allow {
    inside_valid_time_window
}

inside_valid_time_window {
    current_time := time.now_ns()
    time_window_min <= current_time
    time_window_max >= current_time
}
```

**Reply by scottc-WellSky** [reply]
I realize this is old, but I'm also having trouble comparing dates in Rego. The solution in this answer incorrectly returns true for me in the Rego Playground.

**Reply by scottc-WellSky** [reply]
Sorry, this does work correctly when ran as a complete policy. It only returns true when I was evaluating the inside_valid_time_window function by highlighting it and evaluating it in isolation.

**Reply by srenatus** [reply]
It does? 🤔 Curious. It's evaluating to false here: https://play.openpolicyagent.org/p/cbbMARjdET

**Reply by scottc-WellSky** [reply]
Yea, I'm a newbie in this area and apparently don't understand how executing a partial selection of code works in the Playground.
![image](https://github.com/open-policy-agent/community/assets/95659639/d86de7c4-e949-4d75-8852-8481dbfa2969)

**Reply by srenatus** [reply]
Try to only select the rule:
<img width="288" alt="image" src="https://github.com/open-policy-agent/community/assets/870638/c379f82c-64f5-40dc-b376-39b31ac99310">
it'll give you, well, nothing, `undefined`:
<img width="320" alt="image" src="https://github.com/open-policy-agent/community/assets/870638/125233df-d029-4e53-aace-9c6e87ea42eb">

So `allow` will be undefined, too, and thus its default rule hits and `allow` evaluates to false.
